plenum/test/common/test_digest_validation.py

```diff
@@ -95,9 +127,64 @@ def test_second_digest_is_written(
 
     req = Request(**json.loads(req))
 
-    ledger_id, _, full_digest = txnPoolNodeSet[0].seqNoDB.get(req.payload_digest)
+    ledger_id, _ = txnPoolNodeSet[0].seqNoDB.get_by_payload_digest(req.payload_digest)
     assert ledger_id == DOMAIN_LEDGER_ID
-    assert full_digest == req.digest
 
-    payload_digest = txnPoolNodeSet[0].seqNoDB.get(req.digest, full_digest=True)
+    payload_digest = txnPoolNodeSet[0].seqNoDB.get_by_full_digest(req.digest)
     assert payload_digest == req.payload_digest
+
+
+def test_suspicious_primary_send_same_request_with_same_signatures(
```

Comment:
Maybe we can make this test more simple as follows:
- send a request (as usual), make sure that it's ordered
- re-add the request to primary's `requestQueues` (just call `readyFor3PC`?)
- path primary's validation as in `test_suspicious_primary_send_same_request_with_different_signatures`
- make sure that the primary send another 3PC batch with already ordered request
- make sure that this request wasn't ordered, and view has been changed.

Reply:
ok, rewritten

Reply:
fixed